### PR TITLE
Feature/button styles

### LINKED ISSUE #217: [S12.1]: Buttons
### Link to Epic

Epic: https://github.com/Open-Study-College/osc/issues/171

### Description

n/a

### Linked Tasks

- [x] #218 
- [x] #219 
- [x] #248 
#598

### LINKED ISSUE #598: Buttons styling/theming
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/217

### Description

n/a

### PR DESCRIPTION
**Pull request by stevenbister**
- Closes #598 
- Closes #217 

## 📝 Description

Adds the styles for the button variations.

## ⛳️ Current behavior (updates)

N/a

## 🚀 New behavior

Adds the styles for the button variations. 
Part of this includes adding a couple of new colour tokens as well as a light font weight to the typography tokens. 
I have added a `forwardRef` to the button component here as I realised it was needed, I also ended up using it myself to set a `min-width` on the `CopyButton` so the button doesn't collapse when the `innerText` changes.

## 📝 Additional Information

There is currently a small bug in running storybook due to a font alias not being found. Matt has added a fix to his [PR here](https://github.com/Open-Study-College/osc/pull/630) so I've left it out of this one.

To view the buttons for now you will need to pull this down, move the fonts out of the ImageGallery component and put them into the `fonts` directory before running `npm run storybook`

### CODE REVIEW

**Comment by hi-im-si** [comment]
Looks like before we can close this, there are now a few more variations on the button component

**Comment by stevenbister** [comment]
> Looks like before we can close this, there are now a few more variations on the button component

Sure, I'll have a look at getting these added to this PR. 

Any thoughts on naming for them? I think we're up to octonary already, shall I just continue with that sequence? 
But then these new ones would be the "primary" button on the academic hub :S

**Comment by stevenbister** [comment]
Gonna set this to draft for the time being until Amy has consolidated the buttons

**Review by hi-im-si — CHANGES_REQUESTED** [review]

File: packages/osc-ecommerce/app/types/sanity.ts [hunk 1]
```diff
@@ -51,6 +51,14 @@ interface buttonModule extends module {
...
     reference?: object;
     type: string;
+    variant?:
+        | 'primary'
+        | 'secondary'
+        | 'tertiary'
+        | 'quaternary'
+        | 'quinary'
+        | 'septenary'
+        | 'octonary';
```
[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-ecommerce/app/types/sanity.ts]: All done 👍

File: packages/osc-studio/schemas/objects/module/button.ts [hunk 2]
```diff
@@ -11,7 +11,26 @@ export default {
...
+            type: 'string',
+            options: {
+                list: [
+                    'primary',
+                    'secondary',
+                    'tertiary',
+                    'quaternary',
+                    'quinary',
+                    'septenary',
+                    'octonary',
```
[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-studio/schemas/objects/module/button.ts]: All done 👍

**Pull request merged**